common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -349,6 +349,28 @@ static boolean equals(byte[] bytes1, int startPos1, int endPos1, byte[] bytes2,
         return bytes1[startPos1] == bytes2[startPos2];
     }
 
+    static int hashCode(byte[] bytes, int startPos, int endPos, long prime) {
```

Comment:
@Scottmitch 

Oh, so now you changed the hash function :-). How can you proof that this new hash function produces a better hash code or at least isn't likely to have more collisions? It might evaluate faster, but it might also produce a worse hash code. That's why I was hesitant to change the hash function with my equals performance update.

Looking at `j.l.String` and Hotspot, they are calculating the hash code character by character and it's not intrinsified by HotSpot. I think that's also what we should do. The hashcode is being cached after all and as I said in another comment, I haven't seen `AsciiString.hashCode()` show up in any profiler while working on the headers.